porch/engine/pkg/engine/builtinruntime.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package engine
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/apply-replacements/replacements"
+	"github.com/GoogleContainerTools/kpt-functions-catalog/functions/go/set-namespace/transformer"
+	fnsdk "github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	v1 "github.com/GoogleContainerTools/kpt/pkg/api/kptfile/v1"
+	"github.com/GoogleContainerTools/kpt/pkg/fn"
+	"github.com/GoogleContainerTools/kpt/porch/engine/pkg/kpt"
+)
+
+type builtinRuntime struct {
+	fnMapping map[string]fnsdk.ResourceListProcessor
+}
+
+func newBuiltinRuntime() *builtinRuntime {
+	return &builtinRuntime{
+		fnMapping: map[string]fnsdk.ResourceListProcessor{
+			"gcr.io/kpt-fn/apply-replacements:unstable":                                                                fnsdk.ResourceListProcessorFunc(replacements.ApplyReplacements),
+			"gcr.io/kpt-fn/apply-replacements@sha256:c7a1b2d96e7f02e8aca6b1d15c15542c73f5a9f072c7fc6a3569160f8662e056": fnsdk.ResourceListProcessorFunc(replacements.ApplyReplacements),
+			"gcr.io/kpt-fn/set-namespace:unstable":                                                                     fnsdk.ResourceListProcessorFunc(transformer.SetNamespace),
+			"gcr.io/kpt-fn/set-namespace@sha256:36bf1a791436a75c0108c071bdb34c2cfe97329d548ac47329119dad87d6c6e2":      fnsdk.ResourceListProcessorFunc(transformer.SetNamespace),
```

Comment:
Just a note for us: so this mapping needs to be updated whenever we updated the vendor function's version. And for the short term, we will only maintain one version of a function.

Reply:
I was wondering what are those `SHA1` are for ? probably testing ?

Reply:
> I was wondering what are those SHA1 are for ? probably testing ?

Some users prefer to use SHA instead of tag. Add the SHA here ensures we can serve these users.

We will be using released version for these functions. They are not changing as rapidly as `unstable` tag.
e.g. We will use `gcr.io/kpt-fn/set-namespace:v0.3.0` and it SHA when we import set-namespace:v0.3.0 in `go.mod`.

Reply:
Ack.